phost/src/main.rs

```diff
@@ -83,7 +95,28 @@ async fn get_block_at(client: &subxt::Client<Runtime>, h: Option<u32>)
                              .ok_or(Error::BlockNotFound())?;
 
     let block = deopaque_signedblock(opaque_block);
-    Ok(block)
+
+	let block_with_events;
+	if h.is_some() && h.unwrap() > 0 {
+		block_with_events = fetch_events(&client, &block, h.unwrap()).await.unwrap();
+	} else {
+		block_with_events = BlockWithEvents {
+			block,
+			events: None,
+			proof: None,
+			key: None,
+		}
+	}
```

Comment:
```suggestion	
	let block_with_events = if with_events {
		fetch_events(&client, &block, block.block.header.number).await
			.expect("Failed to fetch event");
	} else {
		BlockWithEvents {
			block,
			events: None,
			proof: None,
			key: None,
		}
	}
```

When h is None, it means the caller wants to get the latest finalized block. We still should get the events at the received block, if the event is required.